src/gatsby-ssr.tsx

```diff
@@ -2,7 +2,7 @@ import { GatsbySSR, RenderBodyArgs, WrapRootElementNodeArgs } from 'gatsby';
 import React from 'react';
 import { HelmetProvider, HelmetServerState } from 'react-helmet-async';
 
-const context: { helmet?: HelmetServerState } = {};
+const context: {[pathname: string]: { helmet?: HelmetServerState }} = {};
```

Comment:
Is this correctly typed?

On Line #14 we're checking `if (helmet) ` and then accessing `helmet.base` but it's actually `helmet.helmet.base` now?


```npm ERR! src/gatsby-ssr.tsx(15,38): error TS2339: Property 'base' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(16,39): error TS2339: Property 'title' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(18,20): error TS2339: Property 'priority' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(19,20): error TS2339: Property 'meta' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(20,20): error TS2339: Property 'link' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(21,20): error TS2339: Property 'style' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(22,20): error TS2339: Property 'script' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(23,20): error TS2339: Property 'noscript' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(32,34): error TS2339: Property 'htmlAttributes' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.
npm ERR! src/gatsby-ssr.tsx(33,34): error TS2339: Property 'bodyAttributes' does not exist on type '{ helmet?: HelmetServerState | undefined; }'.

Reply:
Actually this typing is probably correct but we should access context on `onRenderBody` like this: 
```
export const onRenderBody: GatsbySSR['onRenderBody'] = ({
  pathname,
  setHeadComponents,
  setHtmlAttributes,
  setBodyAttributes,
}: RenderBodyArgs): void => {
  const { helmet } = context[pathname]
```